src/network/networkd-netlabel.c

```diff
@@ -0,0 +1,194 @@
+                return address_add_netlabel_set(address, address->link->network->dhcp_netlabels);
+        case NETWORK_CONFIG_SOURCE_DHCP6:
+                return address_add_netlabel_set(address, address->link->network->dhcp6_netlabels);
+        case NETWORK_CONFIG_SOURCE_DHCP_PD:
+                return address_add_netlabel_set(address, address->link->network->dhcp_pd_netlabels);
+        case NETWORK_CONFIG_SOURCE_NDISC:
+                return address_add_netlabel_set(address, address->link->network->ndisc_netlabels);
+        case NETWORK_CONFIG_SOURCE_STATIC:
+                return address_add_netlabel_set(address, address->netlabels);
+        default:
+                return;
+        }
+}
+
+void address_del_netlabel(const Address *address) {
+        int r;
+        _cleanup_free_ char *addr_str = NULL;
+
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network || !IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        (void) in_addr_prefix_to_string(address->family, &address->in_addr, address->prefixlen, &addr_str);
+
+        r = netlabel_command(NLBL_UNLABEL_C_STATICREMOVE, NULL, address);
+        if (r < 0)
+                log_link_warning_errno(address->link, r, "Deleting NetLabels for IP address %s failed, ignoring",
+                                       strna(addr_str));
+        else
+                log_link_debug(address->link, "Deleting NetLabels for IP address %s", strna(addr_str));
+}
+
+int config_parse_netlabel(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        int r;
+        Set **set = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(set);
+
+        if (isempty(rvalue)) {
+                *set = set_free(*set);
+                return 0;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *w = NULL;
+
+                r = extract_first_word(&p, &w, NULL, 0);
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Failed to extract NetLabel label, ignoring: %s", rvalue);
+                        return 0;
+                }
+                if (r == 0)
+                        return 0;
+
+                /* Label semantics depend on LSM but let's do basic checks */
+                if (!string_is_safe(w)) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "Bad NetLabel label, ignoring: %s", w);
```

Comment:
If `w` is not safe, it shouldn't be printed. Please `cescape` it before pritning.